pkg/storage/options.go

```diff
@@ -52,6 +53,14 @@ func WithContentType(contentType string) UploadOption {
 	}
 }
 
+// WithAllowOverwrite sets the overwrite flag to allow overwriting the destination object.
+func WithAllowOverwrite(allowOverwrite bool) UploadOption {
```

Comment:
This is needed because multiple plan runs will need to overwrite the existing plan file in the GCS bucket.